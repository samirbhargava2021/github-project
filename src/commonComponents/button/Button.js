import React from "react";
import "./Button.scss";

const Button = (props) => {
  const {events} = props;
  const {text, theme} = props?.data;
  const {buttonSize} = props?.options;
  return (
    <button
      onClick={events} 
      className={`${buttonSize ==='big' ? 'big-button' : 'button'} ${theme}`}
    >
      {text}
    </button>
  );
};

export default Button;
