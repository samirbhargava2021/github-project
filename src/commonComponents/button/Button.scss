.button {
  padding: 12px 6px;
  cursor: pointer;
  font-weight: 600;
  border: 1px solid #fce4ec;
  width: 120px;
}

.big-button {
  padding: 12px 6px;
  cursor: pointer;
  font-weight: 600;
  border: 1px solid #fce4ec;
  margin: 0 5%;
  width: 90%;
}