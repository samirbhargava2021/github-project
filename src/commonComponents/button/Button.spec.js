import { shallow } from "enzyme";
import { render, screen } from "@testing-library/react";
import Button from "./Button";

describe("ButtonComponent", () => {
  const mockCallBack = jest.fn();
  const props = {
    events: mockCallBack,
    data:{
      theme:"light",
      text:"Search"
    },
    options:{
        buttonSize:"small"
    }
  }
  it("should render Button Component", () => {
    const wrapper = shallow(<Button {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should have text passed for Button", () => {
    render(<Button {...props} />);
    expect(screen.getByText('Search')).toBeInTheDocument();
  });

  it("should handle Button click event", () => {
    const button = shallow(<Button {...props} />);
    button.find("button").simulate("click");
    expect(mockCallBack).toHaveBeenCalled();
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});
