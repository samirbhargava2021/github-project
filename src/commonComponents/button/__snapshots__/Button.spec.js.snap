// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonComponent should render Button Component 1`] = `ShallowWrapper {}`;
